cot-cli/src/main.rs

```diff
@@ -73,44 +96,58 @@ fn main() -> anyhow::Result<()> {
         .init();
 
     match cli.command {
-        Commands::New {
-            path,
-            name,
-            source: cot_source,
-        } => {
-            let project_name = match name {
-                None => {
-                    let dir_name = path
-                        .file_name()
-                        .with_context(|| format!("file name not present: {}", path.display()))?;
-                    dir_name.to_string_lossy().into_owned()
-                }
-                Some(name) => name,
-            };
-
-            let cot_source = if cot_source.use_git {
-                CotSource::Git
-            } else if let Some(path) = &cot_source.cot_path {
-                CotSource::Path(path)
-            } else {
-                CotSource::PublishedCrate
-            };
-            new_project(&path, &project_name, &cot_source)
-                .with_context(|| "unable to create project")?;
+        Commands::New(args) => handle_new_project(args),
+        Commands::Migration(cmd) => match cmd {
+            MigrationCommands::List(args) => handle_migration_list(args),
+            MigrationCommands::Make(args) => handle_migration_make(args),
+        },
+    }
+}
+
+fn handle_new_project(ProjectNewArgs { path, name, source }: ProjectNewArgs) -> anyhow::Result<()> {
+    let project_name = match name {
+        None => {
+            let dir_name = path
+                .file_name()
+                .with_context(|| format!("file name not present: {}", path.display()))?;
+            dir_name.to_string_lossy().into_owned()
         }
-        Commands::MakeMigrations {
-            path,
-            app_name,
-            output_dir,
-        } => {
-            let path = path.unwrap_or_else(|| PathBuf::from("."));
-            let options = MigrationGeneratorOptions {
-                app_name,
-                output_dir,
-            };
-            make_migrations(&path, options).with_context(|| "unable to create migrations")?;
+        Some(name) => name,
+    };
+
+    let cot_source = if source.use_git {
+        CotSource::Git
+    } else if let Some(path) = &source.cot_path {
+        CotSource::Path(path)
+    } else {
+        CotSource::PublishedCrate
+    };
+    new_project(&path, &project_name, &cot_source).with_context(|| "unable to create project")
+}
+
+fn handle_migration_list(MigrationListArgs { path }: MigrationListArgs) -> anyhow::Result<()> {
+    let path = path.unwrap_or_else(|| PathBuf::from("."));
+    let migrations = list_migrations(&path).with_context(|| "unable to list migrations")?;
+    for (app_name, migs) in migrations {
+        for mig in migs {
+            println!("{app_name}\t{mig}");
```

Comment:
I'm not sure if this is the final output we want but should be fine for now.

Reply:
It's good for now